src/esbmc/bmc.cpp

```diff
@@ -917,38 +942,61 @@ smt_convt::resultt bmct::multi_property_check(
         std::ofstream out(fmt::format("{}-{}", ce_counter++, output_file));
         show_goto_trace(out, ns, goto_trace);
       }
-      std::ostringstream oss;
-      log_fail("\n[Counterexample]\n");
-      show_goto_trace(oss, ns, goto_trace);
-      log_result("{}", oss.str());
+      if (options.get_bool_option("result-only"))
+      {
+        std::ostringstream oss;
+        log_fail("\n[Counterexample]\n");
+        show_goto_trace(oss, ns, goto_trace);
+        log_result("{}", oss.str());
+      }
       final_result = result;
 
       // Update fail-fast-counter
       fail_fast_cnt++;
 
       // for kind && incr: remove verified claims
+      // we should not remove it in each job run as it may be
+      // the same claim which claim_msg with different claim_msg due to unwinding
       if (is_clear_verified)
       {
-        for (auto &it : symex->goto_functions.function_map)
+        // make copy
+        std::string _msg = claim.claim_msg;
+        std::string _loc = claim.claim_loc;
+        to_remove_claims.insert(std::make_pair(_msg, _loc));
+      }
+    }
+  };
+
+  std::for_each(std::begin(jobs), std::end(jobs), job_function);
+
+  // remove verified claims from the Goto functions
+  if (is_clear_verified && !to_remove_claims.empty())
+  {
+    Forall_goto_functions (f_it, symex->goto_functions)
+    {
+      if (f_it->second.body_available)
+      {
+        goto_programt &goto_program = f_it->second.body;
+        std::string claim_msg, claim_loc;
+        std::pair<std::string, std::string> claim_pair;
+        Forall_goto_program_instructions (it, goto_program)
         {
-          for (auto &instruction : it.second.body.instructions)
+          if (it->is_assert())
           {
-            if (
-              instruction.is_assert() &&
-              from_expr(ns, "", instruction.guard) == claim.claim_msg &&
-              instruction.location.as_string() == claim.claim_loc)
+            claim_msg = from_expr(ns, "", it->guard);
+            claim_loc = it->location.as_string();
+            claim_pair = std::make_pair(claim_msg, claim_loc);
```

Comment:
Instead of a pair coldn't you just concatenate the strings? That should give a better interface (std::string) instead of (pair<string,string>)

Reply:
Okay I will change to string concatenate